e2e/packages/sync-test/registerDelegationWithSignature.test.ts

```diff
@@ -0,0 +1,110 @@
+import { afterEach, beforeEach, describe, expect, it } from "vitest";
+import type { ViteDevServer } from "vite";
+import { Browser, Page } from "@playwright/test";
+import { createAsyncErrorHandler } from "./asyncErrors";
+import { deployContracts, startViteServer, startBrowserAndPage, openClientWithRootAccount } from "./setup";
+import { rpcHttpUrl } from "./setup/constants";
+import { waitForInitialSync } from "./data/waitForInitialSync";
+import { createBurnerAccount, resourceToHex, transportObserver } from "@latticexyz/common";
+import { http, createWalletClient, ClientConfig } from "viem";
+import { mudFoundry } from "@latticexyz/common/chains";
+import { encodeEntity } from "@latticexyz/store-sync/recs";
+import { callPageFunction } from "./data/callPageFunction";
+import worldConfig from "@latticexyz/world/mud.config";
+import { worldToV1 } from "@latticexyz/world/config/v2";
+import { types } from "@latticexyz/world/internal";
+import { getWorld } from "./data/getWorld";
+import { callRegisterDelegationWithSignature } from "./data/callRegisterDelegationWithSignature";
+
+const PRIVATE_KEY = "0x67bbd1575ecc79b3247c7d7b87a5bc533ccb6a63955a9fefdfaf75853f7cd543";
+
+const worldConfigV1 = worldToV1(worldConfig);
+
+describe("registerDelegationWithSignature", async () => {
+  const asyncErrorHandler = createAsyncErrorHandler();
+  let webserver: ViteDevServer;
+  let browser: Browser;
+  let page: Page;
+
+  beforeEach(async () => {
+    asyncErrorHandler.resetErrors();
+
+    await deployContracts(rpcHttpUrl);
+
+    // Start client and browser
+    webserver = await startViteServer();
+    const browserAndPage = await startBrowserAndPage(asyncErrorHandler.reportError);
+    browser = browserAndPage.browser;
+    page = browserAndPage.page;
+  });
+
+  afterEach(async () => {
+    await browser.close();
+    await webserver.close();
+  });
+
+  it("can generate a signature and register a delegation", async () => {
+    await openClientWithRootAccount(page);
+    await waitForInitialSync(page);
+
+    // Set up client
+    const clientOptions = {
+      chain: mudFoundry,
+      transport: transportObserver(http(mudFoundry.rpcUrls.default.http[0] ?? undefined)),
+    } as const satisfies ClientConfig;
+
+    const account = createBurnerAccount(PRIVATE_KEY);
+    const walletClient = createWalletClient({
+      ...clientOptions,
+      account,
+    });
+
+    const worldContract = await getWorld(page);
+
+    // Sign registration message
+    const delegatee = "0x7203e7ADfDF38519e1ff4f8Da7DCdC969371f377";
```

Comment:
where is this address coming from? is it `account.address`?

Reply:
It's just a made up address! it doesn't matter who the delegatee is. 

I'll make this more clear

Reply:
This example is confusing because there are three accounts:
- `delegator`, which is generated during the test and signs the message
- `delegatee`, which is just generated during the test to fill in the delegatee value
- `walletClient.account.address`, which actually registers the delegation onchain but is otherwise not involved

Reply:
ohh i see